app/parse.py

```diff
@@ -1,15 +1,64 @@
+import csv
+import requests
+from bs4 import BeautifulSoup
 from dataclasses import dataclass
+from typing import List
+
+
+BASE_URL = "https://quotes.toscrape.com/"
 
 
 @dataclass
 class Quote:
     text: str
     author: str
-    tags: list[str]
+    tags: List[str]
+
+
+def parse_quote(quote_soup) -> Quote:
+    text = quote_soup.select_one(".text").get_text(strip=True)
+    author = quote_soup.select_one(".author").get_text(strip=True)
+    tags = [tag.get_text(strip=True) for tag in quote_soup.select(".tag")]
+    return Quote(text, author, tags)
+
+
+def get_all_quotes() -> List[Quote]:
+    quotes = []
+    page_num = 1
+
+    while True:
+        response = requests.get(f"{BASE_URL}/page/{page_num}")
+        soup = BeautifulSoup(response.content, "html.parser")
+
+        quote_blocks = soup.select(".quote")
+        if not quote_blocks:
+            break
+        for quote_block in quote_blocks:
+            quote = parse_quote(quote_block)
+            quotes.append(quote)
+
+        next_button = soup.select_one(".next > a")
+        if not next_button:
+            break
+        page_num += 1
+
+    return quotes
+
+
+def save_quotes_to_csv(quotes: List[Quote], output_csv_path: str) -> None:
+    with open(output_csv_path, "w", newline="") as f:
```

Comment:
It's a good practice to handle potential exceptions when opening files. Consider using a try-except block to catch `IOError` or `OSError` to handle cases where the file cannot be opened or written to.